semantic-analysis/src/Analysis/Syntax.hs

```diff
@@ -102,26 +114,28 @@ parseFile path = do
     -- FIXME: this should use the span of the source file, not an empty span.
     Right (_, Just root) -> pure (File (Ref.fromPath (Path.absRel path)) root)
 
-parseGraph :: A.Value -> A.Parser (IntMap.IntMap Term, Maybe Term)
+newtype Graph = Graph { terms :: IntMap.IntMap Term }
+
+parseGraph :: A.Value -> A.Parser (Graph, Maybe Term)
 parseGraph = A.withArray "nodes" $ \ nodes -> do
-  (untied, First root) <- foldMap (\ (k, v, r) -> ([(k, v)], First r)) <$> traverse (A.withObject "node" parseNode) (V.toList nodes)
-  -- @untied@ is a list of key/value pairs, where the keys are graph node IDs and the values are functions from the final graph to the representations of said graph nodes. Likewise, @root@ is a function of the same variety, wrapped in a @Maybe@.
+  (untied, First root) <- fold <$> traverse parseNode (V.toList nodes)
+  -- @untied@ is an intmap, where the keys are graph node IDs and the values are functions from the final graph to the representations of said graph nodes. Likewise, @root@ is a function of the same variety, wrapped in a @Maybe@.
```

Comment:
I rearranged where some of the types are packed/unpacked in an early attempt to ingest function names without making identifiers out of them. I eventually decided to match the name node directly and grab its source text for the string. (🎩 @dcreager for the suggestion.)